cli/types/cypress.d.ts

```diff
@@ -1421,6 +1417,29 @@ declare namespace Cypress {
      */
     off: Actions
 
+    /**
+     * Enables running Cypress commands in a secondary origin.
+     * @see https://on.cypress.io/origin
+     * @example
+     *    cy.origin('example.com', () => {
+     *      cy.get('h1').should('equal', 'Example Domain')
+     *    })
+     */
+    origin(urlOrDomain: string, fn: () => void): Chainable
+
+    /**
+     * Enables running Cypress commands in a secondary origin.
+     * @see https://on.cypress.io/origin
+     * @example
+     *    cy.origin('example.com', { args: { key: 'value', foo: 'foo' } }, ({ key, foo }) => {
+     *      expect(key).to.equal('value')
+     *      expect(foo).to.equal('foo')
+     *    })
+     */
+    origin<T>(urlOrDomain: string, options: {
+      args: T
+    }, fn: (args: T) => void): Chainable
```

Comment:
```suggestion
    origin<T, S extends any>(urlOrDomain: string, options: {
      args: T
    }, fn: (args: T) => S): Chainable<S>
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21137/commits/0c363dfeb095b20833a38d369e89924e229c4da1